### PR TITLE
Fix overlay rendering in HMD

### PR DESCRIPTION
**Pull request by jherico**
This restores the overlay functionality in the HMD, which has been broken since the team teaching merge.  It now uses the GPU pipeline for the compositing of the overlay with the scene (though not yet for the rendering of overlay contents themselves).

It does not currently support the magnifier functionality, and the stereo overlay has been changed to live precisely in screen space (i.e. the depth of the overlay when working with a 3D monitor will be precisely the distance from the user to the screen).  

It also changes the default cursor to a traditional arrow, rather than the circle reticle.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: bc206633f5b1916a009fad216ee56332065349d3 available at: https://gist.github.com/385c3095b20d4e32f082<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5106.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5106.dmg

File: interface/src/ui/ApplicationOverlay.cpp [hunk 1]
```diff
@@ -236,259 +239,174 @@ void ApplicationOverlay::renderOverlay(RenderArgs* renderArgs) {
...
-            _overlays.buildVBO(_textureFov, _textureAspectRatio, 80, 80);
-        }
+    // Get the mouse coordinates and convert to NDC [-1, 1]
+    vec2 mousePosition = vec2(qApp->getMouseX(), qApp->getMouseY());
+    mousePosition /= canvasSize;
+    mousePosition *= 2.0f;
+    mousePosition -= 1.0f;
+    mousePosition.y *= -1.0f;
+    model.setTranslation(vec3(mousePosition, 0));
+    glm::vec2 mouseSize = 32.0f / canvasSize;
```
Comment (ZappoMan) [inline interface/src/ui/ApplicationOverlay.cpp]: magic number! where does this 32 come from??

**Comment by hifi-gustavo** [comment]
Build console output for commit: d3cdbc389a4c7d3fda287ebe4ac440b0ccb4e051 available at: https://gist.github.com/385c3095b20d4e32f082<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5106.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5106.dmg

File: interface/src/ui/ApplicationOverlay.cpp (hunk 1 above)
Comment (jherico) [inline interface/src/ui/ApplicationOverlay.cpp]: when an 8 and a 4 love each other very much...

fixed

**Comment by hifi-gustavo** [comment]
Build console output for commit: 92f22bc6d52bde359f7b6fc8baa0678e139ba311 available at: https://gist.github.com/385c3095b20d4e32f082<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5106.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5106.dmg

**Pull request merged**